crates/net/network/src/transactions/fetcher.rs

```diff
@@ -44,31 +44,31 @@ use super::{
 #[pin_project]
 pub struct TransactionFetcher {
     /// All peers with to which a [`GetPooledTransactions`] request is inflight.
-    pub active_peers: LruMap<PeerId, u8, ByLength>,
+    pub(super) active_peers: LruMap<PeerId, u8, ByLength>,
     /// All currently active [`GetPooledTransactions`] requests.
     ///
     /// The set of hashes encompassed by these requests are a subset of all hashes in the fetcher.
     /// It's disjoint from the set of hashes which are awaiting an idle fallback peer in order to
     /// be fetched.
     #[pin]
-    pub inflight_requests: FuturesUnordered<GetPooledTxRequestFut>,
+    pub(super) inflight_requests: FuturesUnordered<GetPooledTxRequestFut>,
     /// Hashes that are awaiting an idle fallback peer so they can be fetched.
     ///
     /// This is a subset of all hashes in the fetcher, and is disjoint from the set of hashes for
     /// which a [`GetPooledTransactions`] request is inflight.
-    pub hashes_pending_fetch: LruCache<TxHash>,
+    pub(super) hashes_pending_fetch: LruCache<TxHash>,
     /// Tracks all hashes in the transaction fetcher.
     pub(super) hashes_fetch_inflight_and_pending_fetch: LruMap<TxHash, TxFetchMetadata, Unlimited>,
     /// Filter for valid announcement and response data.
     pub(super) filter_valid_message: MessageFilter,
     /// Info on capacity of the transaction fetcher.
-    pub info: TransactionFetcherInfo,
+    pub(super) info: TransactionFetcherInfo,
     /// [`FetchEvent`]s as a result of advancing inflight requests. This is an intermediary ¨
     /// storage, before [`TransactionsManager`](super::TransactionsManager) streams them.
     #[pin]
-    pub fetch_events_head: UnboundedMeteredReceiver<FetchEvent>,
+    pub(super) fetch_events_head: UnboundedMeteredReceiver<FetchEvent>,
     /// Handle for queueing [`FetchEvent`]s as a result of advancing inflight requests.
-    pub fetch_events_tail: UnboundedMeteredSender<FetchEvent>,
+    pub(super) fetch_events_tail: UnboundedMeteredSender<FetchEvent>,
```

Comment:
there's no need for a sync primitive here if we're using this as a vecdeque, right?

Reply:
a `VecDeque` is slightly more heavy since it keeps track of its length and links the queue in two directions. I think channels don't exclusively need to be used for async communication

Reply:
hmm, vecdeque is a ringbuffer, popping a value is just reading+shifting the head.
channel has a bunch of sync overhead that we don't need because nothing is shared here